The-Chonky-Panda/Atelier
# Interactions wrapper

## Description
Created a HOC with an event listener that we can wrap around our highest level components (e.g. `OverviewContainer`). Added a `postInteractions` action that makes the post request to the server, and then a post handler in our server that makes the call to the api. Clicking on any element in a widget (as long as the main component is wrapped) should get a 201 back from the api that's logged in the server and client.

## How to test
Click on any element in the product overview widget, open up the console and make sure that a 201 is logged.

## Notes
- Make sure that the highest level element in your widget has an id equal to the name of the widget. So in `OverviewContainer` the top `<div>` has an id of 'product-overview'.
- Pass in a string equal to that id as the second argument when wrapping your widget component in `Wrapper.jsx`
- I also wrapped the `RelatedProducts` component just to make sure our app's functionality was maintained when clicking on a related product. As far as I can tell everything still works but more testing is needed
- Not sure if React fragments can have id attributes, if not I'd recommend changing to `<div>`
- Any element without a `className` attribute won't be able to provide an element parameter to the post request. Doesn't really effect anything so probably doesn't matter, but something to keep in mind
- Performance doesn't seem effected as no change in lighthouse metrics on my end, but not 100% certain
- Also added a lang attribute to our `<html>` since apparently that increased our accessibility score by 6 points lol

## Issue tracking
https://trello.com/c/TRdwLjUK/136-create-interactions-tracker-hoc
